doc/devel/coding_style_guideline.rst

```diff
@@ -49,6 +49,182 @@ No alias should be used for `h5py`::
+The `cimport`s should add the `c` prefix to the usual Python import package
+shorthand, e.g.::
+
+  cimport numpy as cnp
+
+Adding the `c` prefix to the import line makes it clear that the Cython/C
+symbols are being referred to as to compared to the Python symbols.
+
+Variable declaration
+--------------------
+Separate `cdef`, `cpdef`, and `ctypedef` statements from the following type by
+exactly one space. In turn, separate the type from the variable name by exactly
+one space. Declare only one `ctypedef` variable per line. You may `cdef` or
+`cpdef` multiple variables per line as long as these are simple declarations;
+note that multiple assignment, references, or pointers are not allowed on the
+same line. Grouping `cdef` statements is allowed. For example::
+
+  # Good
+  cdef int n
+  cdef char* s
+  cdef double Xf[3]
+  cdef double d[3]
+  cpdef int i, j, k
+  cdef Material mat = Material()
+  cdef:
+      double *ps = <double *> cnp.PyArray_DATA(seed)
+      cnp.npy_intp *pstr = <cnp.npy_intp *> qa.strides
+      cnp.npy_intp d, i, j, cnt
+      double direction[3]
+      double tmp, ftmp
+  cdef int true_enough(x):
+      return 1
+
+  # Bad
+  cdef  char *s
+  cdef char * s, * t, * u, * v
+  cdef double Xf[3], d[3]
+  cdef double x=42, y=x+1, z=x*y
+  cdef Material     mat   = Material()
+  cdef   int   falsified(x):
+      return 0
+
+Inside of a function, place all `cdef` statements at the top of the function
+body::
+
+  # Good
+  cdef int true_enough(x):
+      cdef int i = x
+      cdef int rtn
+      rtn = i + 42
+      return rtn
+
+  # Bad
+  cdef int falsified(x):
+      cdef int i = x, j = -42
+      j += i
+      cdef int rtn = j / j - 1
+      return rtn
+
+Using C libraries
+-----------------
+The `cimport`s should follow the same rules defined in PEP8 for `import`
+statements. If a module is both *imported* and *cimported*, the `cimport`
+should come before the `import`.
+
+An example of an imported C library::
+
+  from libc.stdlib cimport calloc, realloc, free
+
+Do not use `include` statements.
+
+Error return values
+-------------------
+When declaring an error return value with the `except` keyword, use one space on
+both sides of the `except`. If in a function definition, there should be no
+spaces between the error return value and the colon `:`. Avoid `except *`
+unless it is needed for functions returning `void`::
+
+  # Good
+  cdef void redwood() except *
+  cdef void c_extract(Feature self, Data2D datum, Data2D out) nogil except *:
+  cdef int sequoia(x) except +:
+      ...
+
+  # Bad
+  cdef char * spruce(x) except *:
+  cdef int fir(x)    except   +  :
+      ...
+
+Pointers and references
+-----------------------
+Pointers and references may be either zero or one space away from the type name.
+If followed by a variable name, they must be one space away from the variable
+name. Do not put any spaces between the reference operator `&` and the variable
+name::
+
+  # Good
+  cdef int& i
+  cdef char * s
```

Comment:
As said earlier in the PR, personally, I find clearer
```
cdef int &i
```
or
```
cdef char *s
```
but I am aware this may be controversial.

Reply:
Personnally, I prefer just this:

```
cdef int& i
cdef char* s
```
I find it clearer because it avoid to mix different type on the same line. I see often this 
```
cdef char *s, k, *p, t
```
which is confusing (to see 2 declaration types on the same line)

Reply:
The example you pointed would not be allowed according to the `Variable declaration` section rules, since pointer variables should dwell on a line of their own :sweat_smile: .

Whether the dereference operator goes beside the type or not, as said, is subject to debate. I'd be OK with whatever is decided, would update the PR with it, and would adapt to it in DIPY.

I find that whatever is decided, the `Casting` rule/example should also abide by it: i.e. `<void *> s` or `<void*> s`.